modules/storages/lib/open_project/storages/engine.rb

```diff
@@ -159,12 +159,21 @@ def self.permissions
           User.current.allowed_in_project?(:view_file_links, project)
           project.project_storages.each do |project_storage|
             storage = project_storage.storage
-            href = "/api/v3/project_storages/#{project_storage.id}/open"
-            icon = if storage.provider_type_nextcloud?
-                     'nextcloud-circle'
-                   else
-                     'hosting'
-                   end
+            next unless storage.configured?
+
+            url_helpers = Rails.application.routes.url_helpers
+            open_project_storage_url = url_helpers.open_project_storage_url(
```

Comment:
🟡🟢 I don't get, why we have these URL data in some completely unrelated file then what they belong to. I know that this comes from :railway_track: but it would be better placed closes to the storage models.

Reply:
Yes, the piece of logic is kinda big here. We could think together how to improve it or leave it as it is for now.